### PR TITLE
[FEATURE] Enable the passing of an existing suite to `RuleBasedProfiler.run()`

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Allow users to pass in an existing suite to `run()`
- In enabling this, we can now pass the `Validator`'s suite attribute to the method (CLI flow to come in a subsequent PR).

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/data_context/data_context.py
Comment (donaldheppner) [inline great_expectations/data_context/data_context.py:3378]: I love this doc string!

File: great_expectations/rule_based_profiler/rule_based_profiler.py
Comment (donaldheppner) [inline great_expectations/rule_based_profiler/rule_based_profiler.py:338]: I think we can safely `and` these conditions; it should improve readability

File: great_expectations/validator/validator.py
Comment (donaldheppner) [inline great_expectations/validator/validator.py]: if expectation suite has been "private" all this time, do we want to expose it now?

File: tests/rule_based_profiler/test_rule_based_profiler.py
Comment (donaldheppner) [inline tests/rule_based_profiler/test_rule_based_profiler.py:956]: would the default here not suffice?

**Review by donaldheppner — CHANGES_REQUESTED** [review]

File: tests/rule_based_profiler/test_rule_based_profiler.py
Comment (donaldheppner) [inline tests/rule_based_profiler/test_rule_based_profiler.py:1379]: shouldn't we test the functionality? That we provide a suite with at least one expectation and we get back a suite that contains that original expectation plus ones that were added via profiling?

**Review by donaldheppner — CHANGES_REQUESTED** [review]

File: great_expectations/rule_based_profiler/rule_based_profiler.py
Comment (cdkini) [reply great_expectations/rule_based_profiler/rule_based_profiler.py:338]: The possible inputs here are as follows:
```
suite + suite_name       -> AssertionError
suite + no suite_name    -> Skip this whole conditional
no suite + suite_name    -> Create a suite with the given name
no suite + no suite_name -> Create a suite with a generated name
```

In that third case, I think we require this nested set of conditionals right? I don't have a suite but I do have a name - using `and` would skip the logic to create a suite and leave us in a situation where `expectation_suite` is `None`.

File: great_expectations/validator/validator.py
Comment (cdkini) [reply great_expectations/validator/validator.py]: you're right! I want to expose a read-only value through the `@property` but I think internal usage should remain consistent. Changing with this next commit.

File: tests/rule_based_profiler/test_rule_based_profiler.py
Comment (cdkini) [reply tests/rule_based_profiler/test_rule_based_profiler.py:956]: Unfortunately not - although the default is indeed `None`, `call_args` on a mock required very explicit assertions. Since this profiler run was invoked with this kwarg (albeit implicitly), we need to note it in our assertion.

File: tests/rule_based_profiler/test_rule_based_profiler.py
Comment (cdkini) [reply tests/rule_based_profiler/test_rule_based_profiler.py:1379]: you make a very valid point - I do like this test for validating that we're mutating and existing object but will follow with a true test of the actual functionality.

File: great_expectations/rule_based_profiler/rule_based_profiler.py
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/rule_based_profiler.py:338]: @cdkini @donaldheppner A similar logic should already exist in `Validator` -- so can be adapted herein as well (if needed, or defer to `Validator` to perform the checks).  Thanks.

**Review by alexsherstinsky — APPROVED** [review]
LGTM (assuming @donaldheppner 's recommendations have been considered/effected).  Very useful enhancements!  Thanks @cdkini !

File: great_expectations/rule_based_profiler/rule_based_profiler.py
Comment (donaldheppner) [reply great_expectations/rule_based_profiler/rule_based_profiler.py:338]: oh yeah, I see it now

**Review by donaldheppner — APPROVED** [review]

File: great_expectations/data_context/data_context.py
Comment (Shinnnyshinshin) [inline great_expectations/data_context/data_context.py:3395]: Seconded. Thank you very very much @cdkini

File: tests/integration/profiling/rule_based_profilers/test_profiler_basic_workflows.py
Comment (Shinnnyshinshin) [inline tests/integration/profiling/rule_based_profilers/test_profiler_basic_workflows.py:357]: ❤️

**Review by Shinnnyshinshin — APPROVED** [review]
Omg this is amazing @cdkini. Thank you

**Pull request merged**
